qt/python/mantidqt/mantidqt/widgets/workspacedisplay/table/presenter.py

```diff
@@ -133,16 +132,12 @@ def supports(cls, ws):
         return TableWorkspaceDisplayModel.supports(ws)
 
     def replace_workspace(self, workspace_name, workspace):
-        if self.presenter.model.workspace_equals(workspace_name):
-            # stops triggering itemChanged signal while the data is being reloaded
+        if self.presenter.model.workspace_equals(workspace_name) and self.presenter.model.block_model_replace:
```

Comment:
Can I check the logic here? Do we not want to only run the `if` block when `self.presenter.model.block_model_replace` is False. As the variable is called block it sounds like it should be blocking and **not** doing the replacement?

Reply:
yes, sorry, I saw block and thought "yes its blocking signals". 🤦 

I will change it tomorrow.